lib/ash_authentication_phoenix/controller.ex

```diff
@@ -227,4 +231,24 @@ defmodule AshAuthentication.Phoenix.Controller do
     |> Helpers.revoke_session_tokens(otp_app)
     |> Plug.Conn.clear_session()
   end
+
+  @doc """
+  Set the live socket id so we can send disconnects from the server when the token is revoked.
+
+  This ensures that the user can't use the application with revoked tokens.
+  """
+  def set_live_socket_id(conn, token) do
+    with {:ok, claims} <- Jwt.peek(token),
+         otp_app <- conn.private.phoenix_endpoint.config(:otp_app),
+         {:ok, resource} <- Jwt.token_to_resource(token, otp_app),
+         {:ok, token_resource} <-
+           AshAuthentication.Info.authentication_tokens_token_resource(resource),
+         {:ok, template_fn} <-
+           Info.token_live_socket_id_template(token_resource) do
+      conn
+      |> Plug.Conn.put_session(:live_socket_id, template_fn.(claims))
```

Comment:
Do we need to handle failure here? We're not verifying the function, it's arity or the return value.

Reply:
Right but what if a user provides a function that raises an exception or has the wrong arity or returns a nonsense value?

Reply:
We could specify that the function should return a tuple, but this still doesn't solve the default value issue.

But if the user like on purpose returns the same livesocket I'd for all users then that's not our issue. As it will just reconnect the websockets [the token would still be valid] so it will pass through